### PR TITLE
Add pad to torchx

### PR DESCRIPTION
**Pull request by howard0su**

### CODE REVIEW

File: torchx/lib/torchx/backend.ex [hunk 1]
```diff
@@ -717,6 +717,24 @@ defmodule Torchx.Backend do
     {p, l, u}
   end
 
+  @impl true
+  def pad(out, tensor, constant, config) do
+    config =
+      config
+      |> Enum.map(fn {a, b, c} ->
+        if a < 0 || b < 0 || c != 0, do: raise("{#{a}, #{b}, #{c}} padding is not supported.")
```
[the remark this answers is not in the document]
Comment (polvalente) [reply torchx/lib/torchx/backend.ex:725]: What would c != 0 imply?

[the remark this answers is not in the document]
Comment (howard0su) [reply torchx/lib/torchx/backend.ex:725]: torchx doesn't support interleave padding

File: torchx/lib/torchx/backend.ex [hunk 2]
```diff
@@ -717,6 +717,24 @@ defmodule Torchx.Backend do
...
+        [a, b]
+      end)
+      |> List.flatten()
+
+    constant = Nx.to_number(constant)
```
[the remark this answers is not in the document]
Comment (howard0su) [reply torchx/lib/torchx/backend.ex:730]: anyway we need convert. I feel do the converting in elixir is easier than C++.

File: torchx/lib/torchx/backend.ex (hunk 1 above)
Comment (josevalim) [inline torchx/lib/torchx/backend.ex:725]: When all values are boolean, you can use `or` as a strictly boolean operator!

```suggestion
        if a < 0 or b < 0 or c != 0 do
          raise ArgumentError, "{#{a}, #{b}, #{c}} padding is not supported"
        end

```

**Comment by josevalim** [comment]
@howard0su you will have to permanently add `pad: 3` to `@inherently_unsupported_doctests` and manually add tests to `test/nx_test.exs` as it is a subset of the Nx functionality. :)

**Review by josevalim — APPROVED** [review]

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Pull request merged**